### DISCUSSION TITLE
Convert from display-p3 to (s)RGB and retain appearance as much as possible?

### DISCUSSION THREAD

**Discussion by benfrain** [ImageMagick/ImageMagick#5694] [Help] [answered]
Unsure if ImageMagick can help here?

We have display-p3 colours, coming out of Sketch app, represented as RGB, and we want to move them to sRGB and keep the converted colour as similar looking as possible.

So, our input would be something like `#137a5a` (but that represents RGB in display-p3 space), and we want to get a `#rrggbb` back in sRGB colour space that looks as close as possible.

Can ImageMagick do this?

**Comment by snibgo** [comment]
Is your input a raster image, ie does it have pixels? Does your input have an embedded ICC colour profile? If it does, you can convert to sRGB like this:
```
magick in.ext -profile sRGB.icc out.ext
```
Use whatever (raster) extensions you need, like PNG or TIFF or whatever. Use any sRGB ICC profile you want. If that file isn't in your current directory, you should provide the full path.

If it doesn't have an embedded profile, assign one first, eg:
```
magick in.ext -profile p3.icc -profile sRGB.icc out.ext
```
(If it already has an embedded p3 profile, the first "-profile p3.icc" is mostly harmless.)

Before "-profile" to convert to sRGB, you can supply an "-intent". Not all ICC profiles support all four intents.

**Reply by benfrain** [reply]
Ah, no, was actually looking to convert direct from hex to color function or vice versa, so input output would just be a string representing that colour.

**Comment by snibgo** [comment, marked as answer]
An image can have just one pixel.

I suggest using "ftxt:" format for input and output. See [ftxt: formatted text](http://im.snibgo.com/fmttxt.htm)

To convert just one string, such as "#abc" from P3 to sRGB, you can do this:

```
echo #abc | magick -define ftxt:format="\H\n" -size 1x1 ftxt:- -profile P3.icc -profile sRGB.icc ftxt:
```
If you have a file of strings (one string per line), perhaps named "in.txt", you can do this:
```
magick -define ftxt:format="\H\n" -size NNx1 ftxt:in.txt -profile P3.icc -profile sRGB.icc ftxt:
```
EDIT: "NN" should be a number, the number of lines in the input file.

**Reply by benfrain** [reply]
That’s great. Thanks
